ClassTranscribeDatabase/CommonUtils.cs

```diff
@@ -49,17 +52,57 @@ public static T BytesToMessage<T>(byte[] bytes)
             return JsonConvert.DeserializeObject<T>(Encoding.UTF8.GetString(bytes));
         }
 
+        private static int seed; // member var only for future debugging /replay options
+        private static Random random;
+        /// <summary>Ensures we have a well seeded RNG even if two processes or two threads  are started at the same time</summary>    
+        private static Random getSharedRandom()
+        {
+            if (random != null) 
+                return random; // Early return
+            // Use something external and trust worthy if it exists
+            const string randomStream = "/dev/urandom";
+            if (File.Exists(randomStream)) {
+                using (BinaryReader reader = new BinaryReader(File.Open(randomStream, FileMode.Open)))
+                {
+                    seed = reader.ReadInt32();
+                }
+            } else
+            {
+                // Fallback
+                seed = Guid.NewGuid().ToString().GetHashCode();
+            }
+            // TODO/REVIEW:Random ctor seed is limited to 32 bits!
+            // Eventually we will need something better
+            random = new Random(seed);
+            return random;
+        }
         public static string RandomString(int length)
         {
-            Random random = new Random();
-            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
-            return new string(Enumerable.Repeat(chars, length)
-              .Select(s => s[random.Next(s.Length)]).ToArray());
+            // TODO/TOREVIEW: C# random is seeded with a time since 1970 at 100ns, so what if two
+            // threads start at the same time, to within the tick resolution?
+            // e.g. https://stackoverflow.com/questions/1785744/how-do-i-seed-a-random-class-to-avoid-getting-duplicate-random-values#
+            // Could this mean that two threads cmight be seeded with the same number
+            Random random = getSharedRandom();
+            // Some filesystems are case insensitive so only use uppercase
+            // Drop 1 and 0 because they are too similar to IO
+            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ23456789";
+            string result;
+            lock (random)
+            {
+                 result = new string(Enumerable.Repeat(chars, length)
+                  .Select(s => s[random.Next(s.Length)]).ToArray());
+            }
+            return result;
         }
 
         public static string GetTmpFile()
         {
-            return Path.Combine(Globals.appSettings.DATA_DIRECTORY, RandomString(8));
+            // Align with python code.
```

Comment:
why not use [GetTempFileName()](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettempfilename?view=netcore-3.1)

Reply:
Good Question. Because the tmp filename often isnt actually temporary!, and it needs to be in the mounted directory. Would need to check how  GetTempFileName() interacts with Docker.
Other parts of the code distinguish between files that have an extension or not.

We need to take a deep dive into this to understand it and then improve it. Made an issue...
https://github.com/classtranscribe/WebAPI/issues/90